PSICover/PSICover.cs

```diff
@@ -164,30 +167,98 @@ class Analyzer {
          for (int i = 0; i < code.Length; i++)
             code[i] = code[i].Replace ('<', '\u00ab').Replace ('>', '\u00bb');
          foreach (var block in blocks) {
+            ulong bID = hits[block.Id];
             bool hit = hits[block.Id] > 0;
-            string tag = $"<span class=\"{(hit ? "hit" : "unhit")}\">";
-            code[block.ELine] = code[block.ELine].Insert (block.ECol, "</span>");
-            code[block.SLine] = code[block.SLine].Insert (block.SCol, tag);
+            string tag = $"<span class=" + (hit ? $"\"hit\" title= \"No.of hits = {bID}\"" : "\"unhit\"") + ">";
+            for (int l = block.SLine; l <= block.ELine; l++) {
+               code[l] = code[l].Insert (code[l].Length, "</span>");
+               code[l] = code[l].Insert (code[l].TakeWhile(char.IsWhiteSpace).Count(), tag);
+            }
          }
+
+         Directory.CreateDirectory ("HTML");
          string htmlfile = $"{Dir}/HTML/{Path.GetFileNameWithoutExtension (file)}.html";
 
          string html = $$"""
             <html><head><style>
+
             .hit { background-color:aqua; }
             .unhit { background-color:orange; }
             </style></head>
+
             <body><pre>
             {{string.Join ("\r\n", code)}}
             </pre></body></html>
             """;
          html = html.Replace ("\u00ab", "&lt;").Replace ("\u00bb", "&gt;");
          File.WriteAllText (htmlfile, html);
       }
+
       int cBlocks = mBlocks.Count, cHit = hits.Count (a => a > 0);
       double percent = Math.Round (100.0 * cHit / cBlocks, 1);
       Console.WriteLine ($"Coverage: {cHit}/{cBlocks}, {percent}%");
    }
 
+   // Generate output HTML summary table 
+   void GenerateSummary () {
+      ulong[] hits = File.ReadAllLines ($"{Dir}/hits.txt").Select (ulong.Parse).ToArray ();
+      var files = mBlocks.Select (a => a.File).Distinct ().ToArray ();
+      Dictionary<string, double> contents = new ();

```

Comment:
Dictionary is not required. You could use list. You are just adding the elements and sorting it.